### ISSUE TITLE
Minimap Section Header Labels Are Really Hard To Read

### ISSUE THREAD

**Issue by pouyakary** [microsoft/vscode#209989]
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 1.88.0
- OS Version: Darwin arm64 23.0.0

The amazing work of @dgileadi on #190759 has been a wish come true for many of us. (Thanks a lot really!). In the past days trying to use it, I have noticed that it lacks readability a lot. I compared the feature to Xcode and what I saw was two things:

1. Xcode's version has bolder font
2. Xcode's version has letter spacing of around 1px 

The combination of the two makes readability a lot more on Xcode's side on even smaller font sizes than of the vscode: 

<img width="196" alt="Screenshot 1403-01-21 at 11 46 01 PM" src="https://github.com/microsoft/vscode/assets/2157285/4d850467-9568-48eb-ac61-22495cdd26e8">

Thanks to the freshness of the feature, I could read the PR and develop both of them for vscode as well, which I will create as a PR right away:

![](https://github.com/microsoft/vscode/assets/2157285/e54df439-90e2-47f5-8edd-db0c18ac3fb2)

**Comment by pouyakary** [comment]
There are also other things, like configuring the color of the labels (which can also make them a lot readable and unreadable, and I'm going to fix them as well)

**Comment by dimateos** [comment]
Not exactly related, as this issue targets the minimap view.

But we could also let the user configure the looks of the section markers in the _editor_:

![image](https://github.com/microsoft/vscode/assets/26868611/ded8b605-2ee3-4bd9-8d9b-0b221e5e19e6)

I used a highlighter extension like https://marketplace.visualstudio.com/items?itemName=fabiospampinato.vscode-highlight (there are many similar)

I gave a custom style to the marker and the title. Also added custom gutter and scrollbar overview rulers (I can post the config, just ask)

Also if you note, I used two different markers to kind of specify hierarchy and matched them with different syles.
This styling could be potentially mixed with custom tags as in https://github.com/microsoft/vscode/issues/209904, for example to achieve this hierarchy and also style customization.

Overall this marker hightlighting customization is similar to what existing extension do, but with built-in support (hence I hope better performance) and minimap renders.

> Could open a separate issue, but seemed related and discussable here
